admin/app/components/solidus_admin/main_nav/item/component.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+# Menu item within a {MainNavComponent}
+class SolidusAdmin::MainNav::Item::Component < SolidusAdmin::BaseComponent
```

Comment:
I'm still trying to figure out the nesting components. It can lead to very lengthy namespaces. For instance, in #5154, we're nesting the `main_nav` component within a `sidebar` component. Should we then name this one `SolidusAdmin::Sidebar::MainNav::Item`? What if we add a main nav item icon component at some point? `SolidusAdmin::Sidebar::MainNav::Item::Icon`?
On top of that, what if we move the main navigation outside the sidebar and into the header at some point? Making no assumptions about the components' structure regarding their namespaces will help us refactor. Trying to find some rules might end up being a hassle to follow. In this case, it might seem clear that a main nav item belongs to a main nav, but it could be less evident in other cases.
Thoughts?

Reply:
I can report the experience and the steps we went through for the last project I used components in.

We started out having components "organized organically", i.e. with no specific rule except individual common sense. That was fine initially but as the number of components grew we had to figure out a better way.

Looking at the components we had it was pretty evident that we had sort of two sets of components:
- page specific components (think the product details for the PDP)
- ui elements (think of the classic button)

We came up with a soft rule for which every component that was used across multiple pages was probably a UI element and we started a `UI::` namespace for them. We also decided to keep all components specific to a page in a single namespace (e.g. `app/components/products` or `app/components/layout`). That brought back `app/components` to a manageable state.

It also made very important to be able to move components around swiftly as we realized that things are evolving and we didn't want to spend time fixing class nesting and constant references or renaming files.

---

With all of that as context I'll say that adopting this principles/rules the tree structure could probably look like this:

a base layout folder: `app/components/solidus_admin/layout` will contain a `nav/` subfolder inside which we can decide to keep things flatten if we want. But personally I don't see issues with deeper nestings, we're far from the limits of Ruby and constant lookups are cached by the VM anyway.

In the same way I think we need to start with less granular components and extract as we see a direct benefit instead of going with a very fine granularity upfront. 

Our need for basic components is a perfect benchmark of the level we need to reach as customizations of the admin will need to build similar interfaces. In other words I cannot say right now if we need a component for a table row or not, but will be almost self evident once we try to build two or three index tables.

Reply:
Thanks for sharing this @elia! I think it might work but I have one concern related to 

> and extract as we see a direct benefit instead of going with a very fine granularity upfront.

which might be harder than in a specific store, as we should handle backward compatibility. Of course, even if we go with the more granular approach, no one ensures us that that's the correct way of structuring components and that they will never change location. What I'd like to understand (even later) is how we want to approach having this flexibility of extracting components when we see they are used in multiple places taking into account backward compatibility, which should be a lean practice to work.

Reply:
@kennyadsl yes, in the end, when we officially release, we'll need the right amount of granularity to allow extendibility.

When I propose we start simple with bigger components I'm referring mostly to the grace time in which the admin is not released or is still experimental. Lowering the cost of renaming and extracting components (through things like 1-level nesting, the generator, etc.) I think we'll be able to achieve it during that time, taking informed (and more confident) decisions based on our direct experience.

Reply:
Thanks, @elia, for your detailed description of the process you followed.

I'm ok with following the `UI` vs. page namespaced convention.

> a base layout folder: app/components/solidus_admin/layout will contain a nav/ subfolder inside which we can decide to keep things flatten if we want. But personally I don't see issues with deeper nestings, we're far from the limits of Ruby and constant lookups are cached by the VM anyway.

For that, I'd go with a flattened structure. FWIW, I'm not concerned about Ruby's limits or performance. As you said, we'll discover the level of granularity along the way, and I'd like to avoid expensive refactors. As in the example I gave above, it's good if I can introduce the `Sidebar` namespace without needing to rename `MainNav` to `Sidebar::MainNav` and `MainNav::Item` to `Sidebar::MainNav::Item`. I also think it makes things more ergonomic. Are you ok with that?